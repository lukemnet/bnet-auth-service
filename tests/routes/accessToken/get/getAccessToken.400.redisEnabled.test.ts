import fastify from 'fastify';
import fastifyRedis from 'fastify-redis-mock';
import BlizzAPI from 'blizzapi';
import server from '../../../../src/index';
import getConfig from '../../../helper';

BlizzAPI.prototype.getAccessToken = () => Promise.resolve('');

describe('/accessToken/get 400 (Redis enabled)', () => {
  const fastifyServer = fastify() as any;

  beforeAll(() => {
    fastifyServer.register(fastifyRedis, {
      host: '127.0.0.1',
      port: '6379',
      password: '',
      enableReadyCheck: true,
      dropBufferSupport: false,
    });
    fastifyServer.register(server, getConfig(true));
  });

  afterAll(() => fastifyServer.close());

  it('returns 400', async () => {
    expect.assertions(1);

    const res = await fastifyServer.inject({
      method: 'GET',
      url: '/accessToken/get',
    });

    expect(res.statusCode).toBe(400);
  });

  it('returns correct response', async () => {
    expect.assertions(1);

    const res = await fastifyServer.inject({
      method: 'GET',
      url: '/accessToken/get',
    });

    expect(res.payload).toMatchSnapshot();
  });

  it('response is not cached', async () => {
    expect.assertions(1);

    await fastifyServer.inject({
      method: 'GET',
      url: '/accessToken/get?refresh=true',
    });

    const cachedResponse = await fastifyServer.redis.get(
      getConfig(true).redis.cacheSegment,
    );

    expect(cachedResponse).toBeFalsy();
  });
});
