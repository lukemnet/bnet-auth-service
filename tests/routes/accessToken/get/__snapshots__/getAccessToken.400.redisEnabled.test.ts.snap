// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/accessToken/get 400 (Redis enabled) returns correct response 1`] = `"{\\"status\\":400}"`;
